### PR TITLE
add operation as part of begincall interface.

### PR DESCRIPTION
**Pull request by junchaowu**
@prashantv

### CODE REVIEW

File: golang/peer.go [hunk 1]
```diff
@@ -211,7 +211,10 @@ func (p *Peer) BeginCall(ctx context.Context, serviceName string, operationName
 		return nil, err
 	}
 
-	call, err := conn.beginCall(ctx, serviceName, callOptions)
+	if callOptions == nil {
+		callOptions = defaultCallOptions
+	}
+	call, err := conn.beginCall(ctx, serviceName, callOptions, operationName)
```
Comment (prashantv) [inline golang/peer.go]: can you delete the check for `callOptions == nil` in `calloptions.go`

File: golang/outbound.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ import (
 const maxOperationSize = 16 * 1024
 
 // beginCall begins an outbound call on the connection
-func (c *Connection) beginCall(ctx context.Context, serviceName string, callOptions *CallOptions) (*OutboundCall, error) {
+func (c *Connection) beginCall(ctx context.Context, serviceName string, callOptions *CallOptions, operation string) (*OutboundCall, error) {
```
Comment (prashantv) [inline golang/outbound.go]: where are you calling writeOperation in this method?

File: golang/outbound.go [hunk 3]
```diff
@@ -87,6 +87,10 @@ func (c *Connection) beginCall(ctx context.Context, serviceName string, callOpti
 	call.createStatsTags(c.commonStatsTags)
 	call.log = c.log.WithFields(LogField{"Out-Call", requestID})
 
+	if callOptions.Format != HTTP {
+		call.commonStatsTags["target-endpoint"] = string(operation)
```
Comment (prashantv) [inline golang/outbound.go]: can you pass the `callOptions` and `operation` into createStatsTags and move this commonStatsTag setter into that method

Comment (junchaowu) [inline golang/outbound.go]: done.

File: golang/peer.go (hunk 1 above)
Comment (junchaowu) [inline golang/peer.go]: deleted

File: golang/outbound.go (hunk 3 above)
Comment (prashantv) [inline golang/outbound.go]: don't see this change - is it pushed?

Comment (junchaowu) [inline golang/outbound.go]: damn. not commit.

File: golang/outbound.go [hunk 4]
```diff
@@ -170,13 +175,18 @@ func (call *OutboundCall) Response() *OutboundCallResponse {
...
 		"target-service": call.callReq.Service,
 	}
 	for k, v := range connectionTags {
 		call.commonStatsTags[k] = v
 	}
+
+	if callOptions.Format != HTTP {
+		call.commonStatsTags["target-endpoint"] = string(operation)
+	}
+
```
Comment (prashantv) [inline golang/outbound.go]: nit: delete blank, go style normally avoids new lines after a `{` and before a `}`

**Comment by prashantv** [comment]
lgtm

**Pull request merged**